examples/most-redis-docker/README.md

```diff
@@ -0,0 +1,124 @@
+running redis locally.  Follow the node instructions to install node and redis.
+
+## Install
+
+### Pre-requisites
+
+1. Install [git](https://help.github.com/articles/set-up-git/)
+2. Clone this repo:
+
+```sh
+git clone git@github.com:mostjs/core.git
+cd examples/most-redis-docker
+```
+
+### Assumptions
+
+The remainder of this document assumes your working directory is
+`most-redis-docker`.  You may have to modify some of the following directions
+if you chose another folder name.
+
+This project uses environment variables.  If you are using the windows command
+prompt, you may have to change some commands slightly to use env vars. For
+example, rather than type `REDIS_MYCHANNEL=my_channel npm run consume`, you
+would type `set REDIS_MYCHANNEL=my_channel&&npm run consume`.
+
+### Option 1: Docker
+
+1.  Install [docker](https://www.docker.com/get-started).  (You will need to
+    create a [DockerHub](https://hub.docker.com) account.)
+2.  Build the docker image for this project:
+
+```sh
+docker build -t most-redis .
+```
+
+### Option 2: node.js
+
+1. Install [node 10.x](https://nodejs.org/en/) or the latest LTS version.
+2. Install the dependencies for this project:
+
+```sh
+npm install
+```
+
+3. Install [redis](https://redis.io/topics/quickstart).
+
+If you already have an instance of redis running, you may edit the docker-
+compose.yml file in the `most-redis-docker` folder.  Change any values for
+`REDIS_HOST` and `REDIS_PORT` according to your redis instance
+
+## Run
+
+When running this example, the publish service periodically sends JSON messages
+to redis, and the consume services receive the JSON messages and transform
+them.  In your terminal(s), you should expect to see informational logs from
+each service as they start up and as they process the JSON messages.
+
+### Option 1: Docker
+
+1.  From the `most-redis-docker` folder, create and run the docker-compose
+    network defined in `docker-compose.yml`:
+
+```sh
+docker-compose up
+```
+
+Use `Ctrl-C` to stop the docker-compose network.
+
+You may also use `docker-compose up -d` and `docker-compose down` to start and
+stop the docker-compose network in the background.
+
+### Option 2: node.js
+
+1. From the `most-redis-docker` folder:
+
+```sh
+REDIS_MYCHANNEL=my_channel npm run consume
+```
+
+2. Open a new terminal or shell, and from the `most-redis-docker` folder type:
+
+```sh
+REDIS_MYCHANNEL=my_channel npm run consume
+```
+
+3. Open a third terminal or shell, and from the `most-redis-docker` folder type:
+
+```sh
+REDIS_MYCHANNEL=my_channel npm run publish
+```
+
+## Experiment!
+
+### Env vars
+
+Both the `consume` and `publish` services accept env vars.  If you are using
+Docker, these may be modified in the `docker-compose.yml` file.  The following
+env vars are used:
+
+- `REDIS_PORT`: integer, default is `6379`
+- `REDIS_HOST`: string, default is `'127.0.0.1'`
+- `REDIS_CHANNEL`: string, required
```

Comment:
If we decide to keep REDIS_CHANNEL as required, what do you think of moving it first, and bolding or italicizing "_required_" to make it painfully obvious?